### PR TITLE
Updated deployment examples for modules/templates

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Updated deployment examples for modules/templates

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

### CODE REVIEW

File: docs/wiki/The library - Module usage.md [hunk 1]
```diff
@@ -19,74 +15,164 @@ This section provides a guideline on how to use the CARML Bicep modules.
...
 New-AzResourceGroup -Name 'ExampleGroup' -Location "Central US"
 
 $inputObject = @{
- DeploymentName    = 'ExampleDeployment'
- ResourceGroupName = 'ExampleGroup'
- TemplateFile      = "$home\ResourceModules\arm\Microsoft.KeyVault\vault\deploy.bicep"
+  DeploymentName    = 'ExampleDeployment'
+  ResourceGroupName = 'ExampleGroup'
+  # Using a local reference
+  TemplateFile      = "$home\ResourceModules\arm\Microsoft.KeyVault\vault\deploy.bicep"
```
Comment (eriqua) [inline docs/wiki/The library - Module usage.md:32]: I think we're missing parameter files for this and all examples below

File: docs/wiki/The library - Module usage.md [hunk 2]
```diff
@@ -19,74 +15,164 @@ This section provides a guideline on how to use the CARML Bicep modules.
...
+<summary><i>Resource Group</i> scope</summary>
 
 ```PowerShell
 New-AzResourceGroup -Name 'ExampleGroup' -Location "Central US"
 
 $inputObject = @{
- DeploymentName    = 'ExampleDeployment'
- ResourceGroupName = 'ExampleGroup'
- TemplateFile      = "$home\ResourceModules\arm\Microsoft.KeyVault\vault\deploy.bicep"
+  DeploymentName    = 'ExampleDeployment'
```
Comment (eriqua) [inline docs/wiki/The library - Module usage.md:29]: Should we avoid errors related to the deployment name and generate one based on the time?

File: docs/wiki/The library - Module usage.md [hunk 3]
```diff
@@ -19,74 +15,164 @@ This section provides a guideline on how to use the CARML Bicep modules.
...
-az group create --name 'ExampleGroup' --location "Central US"
-$inputObject = @(
-    '--name',           'ExampleDeployment',
-    '--resource-group', 'ExampleGroup',
-    '--template-file',  "$home\ResourceModules\arm\Microsoft.KeyVault\vault\deploy.bicep",
-    '--parameters',     'storageAccountType=Standard_GRS',
-)
-az deployment group create @inputObject
+<details>
+<summary><i>Subscription</i> scope</summary>
```
Comment (eriqua) [inline docs/wiki/The library - Module usage.md:44]: I'd suggest to be even more detailed and specify at the beginning of each scope paragraph how the corresponding bicep module targetscope should look like

File: docs/wiki/The library - Module usage.md (hunk 1 above)
Comment (MrMCake) [reply docs/wiki/The library - Module usage.md:32]: Removed them on purpose as they are optional. However, will add them again if you think they proove useful. Linked to the official docs for all the extra parameters

File: docs/wiki/The library - Module usage.md (hunk 2 above)
Comment (MrMCake) [reply docs/wiki/The library - Module usage.md]: Updated

File: docs/wiki/The library - Module usage.md (hunk 3 above)
Comment (MrMCake) [reply docs/wiki/The library - Module usage.md:48]: Added

File: docs/wiki/The library - Module usage.md (hunk 1 above)
Comment (MrMCake) [reply docs/wiki/The library - Module usage.md]: Added

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**